engine/server/engine/enclave_manager/enclave_manager.go

```diff
@@ -139,7 +140,7 @@ func (manager *EnclaveManager) CreateEnclave(
 		return nil, stacktrace.Propagate(err, "An error occurred validating enclave name '%v'", enclaveName)
 	}
 
-	if manager.enclavePool != nil {
+	if !isProduction && manager.enclavePool != nil {
```

Comment:
I think you could pass the `isProduction` value to the `enclavePool.GetEnclave` method and the value could be compared in the `areRequestedEnclaveParamsEqualToEnclaveInThePoolParams` helper to determine if it can be pulled from the pool or not. I mean in order to follow the same pattern that we did for the other enclave arguments, like `apiContainerLogLevel` for instance.

Reply:
I added a todo. But in general I think production enclaves are less sensitive to startup time since they will be long running